### ISSUE TITLE
Add `use-standard-html` rule to enforce HTML Standard

### ISSUE THREAD

**Issue by yeonjuan** [yeonjuan/html-eslint#315]
I'd like to propose a new rule: `use-standard-html`.
This rule aims to enforce compliance with the [HTML Standard](https://html.spec.whatwg.org/), by reporting HTML code that violates the specification. It would help developers catch mistakes that aren't necessarily invalid HTML, but do go against standard, expected document structures or attribute usage.

### Example

- elements

```html
<!-- ❌ Invalid -->
<html>
  <head></head>
  <body></body>
  <div></div> <!-- Not allowed here -->
</html>
```

Only <head> and <body> elements are allowed directly inside <html>.

- attributes

```html
<!-- ❌ Invalid -->
<meta charset="UTF-8" title="My Page" />
```

**Comment by yeonjuan** [comment]
I'm working on a project that makes the HTML specification programmatically accessible in JavaScript and TypeScript. - https://github.com/yeonjuan/html-standard
